packages/eslint-plugin-testing-library-react/src/rules/prefer-expect-query-by.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  AST_NODE_TYPES,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import { createRule } from '../utils';
+
+interface ExtendedExpression extends TSESTree.CallExpression {
+  callee: TSESTree.MemberExpression | TSESTree.Identifier;
+}
+
+function isIdentifier(
+  node: ExtendedExpression['callee']
+): node is TSESTree.Identifier {
+  return node.type === AST_NODE_TYPES.Identifier;
+}
+
+function isMemberExpression(
+  node: ExtendedExpression['callee']
+): node is TSESTree.MemberExpression {
+  return node.type === AST_NODE_TYPES.MemberExpression;
+}
+
+function isUsingWrongQueries(node: TSESTree.Identifier) {
```

Comment:
Reading the code I am wondering if this isn't more `isCandidateWithWrongQuery` cause it depends on the `callee.name` being `expect` too, right?

Reply:
Hmm well this function only checks if the name of the node is either `getBy` or `getAllBy`.